### ISSUE TITLE
GLXBadContextState crash

### ISSUE THREAD

**Issue by totaam** [Xpra-org/xpra#3352]
Server:
```
/usr/bin/xpra start --start=xterm --bind-tcp=0.0.0.0:10000 :10 --daemon=no \
    --video-encoders=ffmpeg -d compress,ffmpeg
```
Client:
```
xpra attach --no-mmap
```

Run `vlc` in the session, play a 1080p video and click to move the position of playback until the crash:
```
2021-11-12 21:30:24,211 gl_paint_planar(0, 'mpeg1', AVImageWrapper-AVFrameWrapper(0x7f8fdc4e4a40)(YUV420P:(0, 0, 1920, 960, 24):3_PLANES), 'default', 33, 1920, 960, 1920, 960, typedict({'csc': 'YUV420P', 'delayed': 1, 'frame': 165, 'flush-encoder': True, 'encoding': 'mpeg1'}), [<function WindowClient._do_draw.<locals>.record_decode_time at 0x7f8f8c12f550>, <bound method ClientWindowBase.after_draw_refresh of GLClientWindow(6 : GLDrawingArea(6, (1920, 1066), YUV420P))>])
2021-11-12 21:30:24,211 glXMakeCurrent: xid=0x5600036, context=<OpenGL.raw.GLX._types.LP_struct___GLXcontextRec object at 0x7f8f944c7dc0>
2021-11-12 21:30:24,212 GLDrawingArea(6, (1920, 1066), YUV420P).update_planar_textures(1920, 960, AVImageWrapper-AVFrameWrapper(0x7f8fdc4e4a40)(YUV420P:(0, 0, 1920, 960, 24):3_PLANES), 'YUV420P')
2021-11-12 21:30:24,212 updating planar textures: 1920x960 YUV420P
2021-11-12 21:30:24,212 set_alignment(1920, 1920, 'Y') GL_UNPACK_ROW_LENGTH=0, GL_UNPACK_ALIGNMENT=8
2021-11-12 21:30:24,212 texture 0: div=(1, 1), rowstride=1920, 1920x960, data=1843200 bytes, upload=zerocopy:memoryview
2021-11-12 21:30:24,213 set_alignment(960, 960, 'U') GL_UNPACK_ROW_LENGTH=0, GL_UNPACK_ALIGNMENT=8
2021-11-12 21:30:24,213 texture 1: div=(2, 2), rowstride=960, 960x480, data=460800 bytes, upload=zerocopy:memoryview
2021-11-12 21:30:24,214 set_alignment(960, 960, 'V') GL_UNPACK_ROW_LENGTH=0, GL_UNPACK_ALIGNMENT=8
2021-11-12 21:30:24,214 texture 2: div=(2, 2), rowstride=960, 960x480, data=460800 bytes, upload=zerocopy:memoryview
2021-11-12 21:30:24,214 GLDrawingArea(6, (1920, 1066), YUV420P).render_planar_update('default', 33, 1920, 960, 1, 1, 0) pixel_format=YUV420P
2021-11-12 21:30:24,214 painting planar update, format YUV420P
2021-11-12 21:30:24,214 GLDrawingArea(6, (1920, 1066), YUV420P).render_planar_update(..) texture_size=(1920, 960), size=(1920, 1066)
2021-11-12 21:30:24,214 glXMakeCurrent: NULL for xid=0x5600036

(Xpra:1412564): Gdk-ERROR **: 21:30:24.215: The program 'Xpra' received an X Window System error.
This probably reflects a bug in the program.
The error was 'GLXBadContextState'.
  (Details: serial 0 error_code 159 request_code 151 (GLX) minor_code 5)
  (Note to programmers: normally, X errors are reported asynchronously;
   that is, you will receive the error a while after causing it.
   To debug your program, run it with the GDK_SYNCHRONIZE environment
   variable to change this behavior. You can then get a meaningful
   backtrace from your debugger if you break on the gdk_x_error() function.)
Trace/breakpoint trap (core dumped)
```

```
(gdb) py-bt
Traceback (most recent call first):
  File "/usr/lib64/python3.9/site-packages/xpra/platform/xposix/gl_context.py", line 81, in __exit__
    if not GLX.glXMakeCurrent(self.xdisplay, 0, null_context):
  File "/usr/lib64/python3.9/site-packages/xpra/client/gl/gl_window_backing_base.py", line 1454, in gl_paint_planar
  File "/usr/lib/python3.9/site-packages/gi/overrides/Gtk.py", line 1680, in main
    return _Gtk_main(*args, **kwargs)
  File "/usr/lib64/python3.9/site-packages/xpra/client/gtk_base/gtk_client_base.py", line 156, in gtk_main
    Gtk.main()
  File "/usr/lib64/python3.9/site-packages/xpra/client/gtk_base/gtk_client_base.py", line 150, in run
    self.gtk_main()
  File "/usr/lib64/python3.9/site-packages/xpra/scripts/main.py", line 1630, in do_run_client
    return app.run()
  File "/usr/lib64/python3.9/site-packages/xpra/scripts/main.py", line 1150, in run_client
    r = do_run_client(app)
  File "/usr/lib64/python3.9/site-packages/xpra/scripts/main.py", line 452, in do_run_mode
    return run_client(script_file, cmdline, error_cb, options, args, mode)
  File "/usr/lib64/python3.9/site-packages/xpra/scripts/main.py", line 413, in run_mode
    return do_run_mode(script_file, cmdline, error_cb, options, args, mode, defaults)
  File "/usr/lib64/python3.9/site-packages/xpra/scripts/main.py", line 140, in main
    return run_mode(script_file, cmdline, err, options, args, mode, defaults)
  File "/usr/bin/xpra", line 10, in <module>
    sys.exit(main(sys.argv[0], sys.argv))
```

**Comment by totaam** [comment]
According to the [glXMakeCurrent](https://www.khronos.org/registry/OpenGL-Refpages/gl2.1/xhtml/glXMakeCurrent.xml) docs:
`GLXBadContextState` is generated if `glXMakeCurrent` is executed between the execution of `glBegin` and the corresponding execution of `glEnd`.

`GLXBadContextState` is also generated if the rendering context current to the calling thread has GL renderer state `GLX_FEEDBACK` or `GLX_SELECT`.

**Comment by totaam** [comment]
**Wow**.

That was a strange one.
A more careful look at the debug output shows what's wrong:
```
GLDrawingArea(6, (1920, 1066), YUV420P).render_planar_update('default', 33, 1920, 960, 1, 1, 0) pixel_format=YUV420P
```
Instead of the `x` coordinate we have `default`!
And that's actually a server bug found in the `nvjpeg` code path.

That's fixed in 34d8e54 (server side), 7ecbc6d will now also validate the coordinates we get from the server.
